dateparser/date.py

```diff
@@ -461,6 +468,18 @@ def date_strings():
                     if self._is_applicable_locale(locale, s):
                         yield locale
 
+        if self.detect_languages_function:
+            detected_languages = self.detect_languages_function(
+                text=date_string, confidence_threshold=self._settings.LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD
+            )
+
+            languages = map_languages(detected_languages) or self._settings.DEFAULT_LANGUAGES
```

Comment:
I think we use ``DEFAULT_LANGUAGES`` without custom language detection in search dates, but not here - I think it's better to be consistent and do the same thing here.